frontend/src/Game.js

```diff
@@ -334,8 +339,27 @@ export default class Game {
         }
     }
 
-    getTowerCount = () => {
-        return this.getAllTowers().length - 2; // subtract two for portal and castle
+    static from(hashedString) {
+        try {
+            const decodedString = atob(hashedString);
+            const json = JSON.parse(decodedString);
+            const { gameMap, level, gold, castleHP } = json;
+
+            if (!gameMap || !level || gold === null || !castleHP) {
+                throw new Error('Invalid game data');
+            }
+
+            const newGame = new Game(GameMap.from(gameMap));
+            newGame.level = level;
+            newGame.gold = gold;
+            newGame.castle.hp = castleHP;
+            newGame.commitTowers();
+
+            return newGame;
+        } catch (error) {
+            console.error(error);
+            return { error };
```

Comment:
To be consistent with `GameMap.from`
```suggestion
            return null;
```